pkg/imgpkg/cmd/push.go

```diff
@@ -236,6 +236,9 @@ func (o *PushOptions) validateBundle(registry ctlimg.Registry) error {
 	if err != nil {
 		return fmt.Errorf("Unmarshalling image lock: %s", err)
 	}
+	if imgLock.Kind != lf.ImagesLockKind {
```

Comment:
should also check apiVersion.

Reply:
I think this was purposeful because there is the idea that we will want to update the apiVersion by changing the domain from `imgpkg.k14s.io/*` to `imgpkg.carvel.dev/*`.
If we add the API check right now we will have to change it.

Reply:
this is new code, why wouldnt be using imgpkg.k14s.io?

separately from that, the whole point of having apiVersion is so that tool can ensure that it's seeing input in format it supports. input today is in `imgpkg.carvel.dev/v1alpha1` format, so we should check on that. if you are saying that it's in both `imgpkg.k14s.io/v1alpha1` to `imgpkg.carvel.dev/v1alpha1` then we should check on it being either of them. (also note there is no such thing as `/*`, has to be explicit version per points above).

Reply:
Sounds good.
Do you think we can merge this PR as is and add a new one to check the API version?

Reply:
whatever makes sense for yall. as long as apiVersion check makes it in, im ok.